src/events/api/views.py

```diff
@@ -7,84 +7,92 @@
 
-    speech_querysets = [
-        TalkProposal.objects.filter_accepted(),
-        SponsoredEvent.objects.all()
-    ]
+    def get(self, request, *args, **kwargs):
+        event_type_string = request.GET.get("event_types")
+        event_types = event_type_string.split(',') if event_type_string else []
+        data = []
+        for event_type, api_view in [
+            ('talk', TalkListAPIView),
+            ('sponsored', SponsoredEventListAPIView),
+            ('tutorial', TutorialListAPIView),
+        ]:
+            if event_type not in event_types or not event_types:
+                continue
+            view = api_view.as_view()
+            event_data = view(request._request, *args, **kwargs).data
+            data.extend(event_data)
 
-    def get(self, request, **kwargs):
-        talks, sponsored_events = self.speech_querysets
+        if data:
+            return Response(data)
+        else:
+            raise Http404
 
-        context = {"request": request}
-        talk_serializer = serializers.TalkListSerializer(
-            talks, many=True, context=context)
-        sponsored_events_serializer = serializers.SponsoredEventListSerializer(
-            sponsored_events, many=True, context=context)
 
-        response = talk_serializer.data + sponsored_events_serializer.data
-        return Response(response)
+class TalkDetailAPIView(RetrieveAPIView):
+    queryset = ProposedTalkEvent.objects.all()
+    serializer_class = serializers.TalkDetailSerializer
 
 
-class TutorialDetailAPIView(RetrieveAPIView):
-    authentication_classes = [TokenAuthentication]
-    permission_classes = [IsAuthenticated]
+class SponsoredEventDetailAPIView(RetrieveAPIView):
+    queryset = SponsoredEvent.objects.all()
+    serializer_class = serializers.SponsoredEventDetailSerializer
 
+
+class TutorialDetailAPIView(RetrieveAPIView):
     queryset = ProposedTutorialEvent.objects.all()
     serializer_class = serializers.TutorialDetailSerializer
 
-    def get_object(self):
-        pk = self.kwargs["pk"]
-        queryset = self.get_queryset()
-        return get_object_or_404(queryset, proposal_id=pk)
 
-
-class TutorialListAPIView(ListAPIView):
+class SpeechDetailAPIView(APIView):
     authentication_classes = [TokenAuthentication]
     permission_classes = [IsAuthenticated]
 
-    queryset = TutorialProposal.objects.filter_accepted()
-    serializer_class = serializers.TutorialListSerializer
+    def get(self, request, *args, **kwargs):
+        event_id = self.kwargs.get('pk')
+        event_type = self.kwargs.get('event_type')
+        if not event_id or not event_type:
+            raise Http404
+
+        for _type, api_view in [
+            ('talk', TalkDetailAPIView),
+            ('sponsored', SponsoredEventDetailAPIView),
+            ('tutorial', TutorialDetailAPIView),
+        ]:
+            if _type == event_type:
+                view = api_view.as_view()
+                return view(request._request, *args, **kwargs)
```

Comment:
I suggest use dict to get the API view model, and pass the request directly.
```suggestion
        api_view_dict = {
            'talk': TalkDetailAPIView,
            'sponsored': SponsoredEventDetailAPIView,
            'tutorial': TutorialDetailAPIView,
        }
        api_view = api_view_dict[event_type]
        view = api_view.as_view()
        return view(request, *args, **kwargs)
```